### ISSUE TITLE
Implementation of Stopgrad

### ISSUE THREAD

**Issue by GZYZG** [PatrickHua/SimSiam#22]
May I a ask you a question, how did you implment stop grad? I viewed the code, but I didn't find it. Thank you!

**Comment by PatrickHua** [comment]
```python
def D(p, z, version='simplified'): # negative cosine similarity
    if version == 'original':
        z = z.detach() # stop gradient
        p = F.normalize(p, dim=1) # l2-normalize 
        z = F.normalize(z, dim=1) # l2-normalize 
        return -(p*z).sum(dim=1).mean()
```
Stop grad is implemented using the detach operation

**Comment by GZYZG** [comment]
OHHHH, I see, Thank you very much!
